src/main/java/com/minecolonies/colony/jobs/JobGuard.java

```diff
@@ -0,0 +1,74 @@
+package com.minecolonies.colony.jobs;
+
+import com.minecolonies.client.render.RenderBipedCitizen;
+import com.minecolonies.colony.CitizenData;
+import com.minecolonies.entity.ai.basic.AbstractAISkeleton;
+import com.minecolonies.entity.ai.citizen.guard.EntityAIMeleeGuard;
+import com.minecolonies.entity.ai.citizen.guard.EntityAIRangeGuard;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Random;
+
+/**
+ * Job class of the guard.
+ */
+public class JobGuard extends AbstractJob
+{
+    /**
+     * The higher the number the lower the chance to spawn a knight. Default: 3, 50% chance.
```

Comment:
Random.getInt(int bounds)
But when not a power of two the distribution can be wonky, especially in small numbers.
That's why I often get is like 1-100 and test for mod 2